File: python/1465.py
from itertools import pairwise
from typing import List

class Solution:
    def maxArea(self, h: int, w: int, horizontalCuts: List[int], verticalCuts: List[int]) -> int:
        MOD = 10 ** 9 + 7
        h_cuts = [0] + sorted(horizontalCuts) + [h]
        v_cuts = [0] + sorted(verticalCuts) + [w]
        h_max = max(y - x for x, y in pairwise(h_cuts))
        v_max = max(y - x for x, y in pairwise(v_cuts))
        return h_max * v_max % MOD

File: go/1465.go
package main

import "sort"

func getMax(size int, cuts []int) int {
	sort.Ints(cuts)
	res := max(cuts[0], size-cuts[len(cuts)-1])
	for i := 1; i < len(cuts); i++ {
		res = max(res, cuts[i]-cuts[i-1])
	}
	return res
}

func maxArea(h int, w int, horizontalCuts []int, verticalCuts []int) int {
	h_max := getMax(h, horizontalCuts)
	v_max := getMax(w, verticalCuts)
	return (h_max * v_max) % 1_000_000_007
}
